src/algebra/monoid_algebra.lean

```diff
@@ -585,6 +617,40 @@ lemma single_mul_single {a₁ a₂ : G} {b₁ b₂ : k} :
 (sum_single_index (by simp only [zero_mul, single_zero, sum_zero])).trans
   (sum_single_index (by rw [mul_zero, single_zero]))
 
+/-- Like `finsupp.map_domain_add`, but for the convolutive multiplication we define in this file -/
+lemma map_domain_mul {α : Type*} {β : Type*} {α₂ : Type*}
+  [semiring β] [add_monoid α] [add_monoid α₂] {x y : add_monoid_algebra β α}
+  (f : add_hom α α₂) :
+  (map_domain f (x * y : add_monoid_algebra β α) : add_monoid_algebra β α₂) =
+    (map_domain f x * map_domain f y : add_monoid_algebra β α₂) :=
+begin
+  simp only [mul_def, map_domain_sum, map_domain_single],
+  -- I can't work out any way to avoid `conv` here
+  conv_rhs {
+    for (finsupp.sum _ _) [2] {
```

Comment:
Note that without https://github.com/leanprover-community/lean/pull/476, it is quite hard to debug proofs inside `for`, as the errors are ignored and manifest only as `simp_rw` failing below.